File: Leetcode/cpp/sol283.cpp


//https://leetcode.com/problems/move-zeroes/description/
class Solution {
public:
    void moveZeroes(vector<int>& nums) {

        int pos = 0;
        // first, move all the non zero numbers to the beginning of nums
        for(int i = 0; i<nums.size(); i++){
            if(nums[i] != 0){
                nums[pos] = nums[i];
                pos++;
            }
        }
        // then, fill zeros from the last position added
        for(int i = pos; i<nums.size(); i++){
            nums[i] = 0;
        }

    }
};

File: Leetcode/python/sol283.py
# https://leetcode.com/problems/move-zeroes/
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        if len(nums) <= 1:
            return nums
        pos = 0
        for i, j in enumerate(nums):
            if j != 0:
                nums[pos] = j
                pos+=1

        for i in range(pos, len(nums)):
            nums[i] = 0




